tools/spec_to_manifest.py

```diff
@@ -0,0 +1,189 @@
+        if verify_yaml(filename):
+            spec = open_and_read_spec(filename)
+            print("Read spec file at " + filename)
+            json_obj = parse_fields(spec)
+            print("Converted fields")
+            write_to_json(json_obj, path)
+            print("Wrote manifest at " + path)
+        else:
+            logging.warning("ERROR: Wrong file type specified")
+
+
+def parse_fields(yaml_obj):
+    # Manifest Version -- mandatory
+    manifest_obj = {"manifest_version": 1}
+
+    # Extension -- mandatory
+    # In this case, only workflows
+    extension_type = yaml_obj.get("extension")
+    extension_obj = {"type": extension_type, extension_type: yaml_obj.get("name") + ".icon"}
+    manifest_obj["extension"] = extension_obj
+
+    # ID -- mandatory
+    # From yaml
+    manifest_obj["id"] = yaml_obj.get("name")
+
+    # Title -- mandatory
+    # From yaml
+    manifest_obj["title"] = yaml_obj.get("title")
+
+    # Description -- Mandatory
+    # from yaml
+    manifest_obj["tagline"] = yaml_obj.get("description")
+
+    # Overview -- mandatory
+    # Tricky extraction of overview from .md
+    # For now, keep as blank
+    manifest_obj["overview"] = ""
+
+    # Key Features -- mandatory
+    # From .md, for now, keep as blank
+    manifest_obj["key_features"] = []
+
+    # Requirements -- Optional
+    # Tricky reading from .md, due to different formatting....
+    # Keep blank for now
+    manifest_obj["requirements"] = []
+
+    # Resources -- optional
+    # Don't know where to pull this info from.....
+    manifest_obj["resources"] = []
+
+    # Version -- optional
+    # Directly from yaml, parsed as string in manifest
+    manifest_obj["version"] = str(yaml_obj.get("version"))
+
+    # Version History....
+    # Should be from .md -- need to parse
+    manifest_obj["version_history"] = []
+
+    # Publisher -- mandatory
+    # "Vendor" field from yaml
+    manifest_obj["publisher"] = yaml_obj.get("vendor")
+
+    # Support -- mandatory
+    # from yaml if R7, where to get info on "community" support?
+    if yaml_obj.get("support") == "rapid7":
+        support_obj = {"type": "publisher", "url": "http://www.rapid7.com/discuss"}
+    else:
+        support_obj = {"type": "community", "url": ""}
+    manifest_obj["support"] = support_obj
+
+    # Rapid7 Products -- mandatory
+    # Products key from yaml, we don't have an understanding of primary/secondary....
+    product_listings = yaml_obj.get("products")
+    r7_products = []
+    for x in range(len(product_listings)):
+        if x == 0:
+            r7_products.append({"name": product_listings[x], "role": "primary"})
+        else:
+            r7_products.append({"name": product_listings[x], "role": "secondary"})
+    manifest_obj["rapid7_products"] = r7_products
+
+    # Required R7 Features -- Required
+    # Different rules for workflows, plugins, etc....Workflows --> need orchestrator
+    manifest_obj["required_rapid7_features"] = ["orchestrator"]
```

Comment:
do something like this so it automatically sets the orcehstrator when needed

if spec is workflow or spec is plugin:
  if 'cloud_ready' not in spec:
     manifest_obj["required_rapid7_features"] = ["orchestrator"]